src/topp/DecoyDatabase.cpp

```diff
@@ -165,173 +186,234 @@ class TOPPDecoyDatabase :
     f.writeStart(out);
     FASTAFile::FASTAEntry entry;
 
-    // Configure Enzymatic digestion
-    // TODO: allow user-specified regex
+    // Create a ProteaseDigestion object for peptide digestion
     ProteaseDigestion digestion;
     String enzyme = getStringOption_("enzyme").trim();
-    if ((input_type == SeqType::protein) && !enzyme.empty())
-    {
-      digestion.setEnzyme(enzyme);
-    }
-    // check if decoy_string is common decoy string (e.g. decoy, rev, ...)
-    String decoy_string_lower = decoy_string;
-    decoy_string_lower.toLower();
-    bool is_common = false;
-    for (const auto& a : DecoyHelper::affixes)
+    if ((input_type == SeqType::protein) && ! enzyme.empty()) { digestion.setEnzyme(enzyme); }
+    
+    
+ // create neighbor peptides for the relevant peptides given in the FASTA from '-in'
+    String neighbor_file = getStringOption_("Neighbor_Search:neighbor_in");
+ // Check if the neighbor file option is provided
+    if (! neighbor_file.empty())
     {
-      if ((decoy_string_lower.hasPrefix(a) && decoy_string_position_prefix) || (decoy_string_lower.hasSuffix(a) && !decoy_string_position_prefix))
+      String out_neighbor = getStringOption_("out_neighbor");
+      FASTAFile neig;
+      neig.writeStart(out_neighbor);
+      double mz_bin_size = getDoubleOption_("Neighbor_Search:mz_bin_size");
+      double min_shared_ion_fraction = getDoubleOption_("Neighbor_Search:min_shared_ion_fraction");
+      double mass_tolerance = getDoubleOption_("Neighbor_Search:mass_tolerance");
+      int missed_cleavages = getIntOption_("Neighbor_Search:missed_cleavages");
+      digestion.setMissedCleavages(missed_cleavages);
```

Comment:
this is now also used for the original code (which requires MC==0). Either copy the object and work on a copy or reset to MC=0 after you are done. The first option is cleaner.

Reply:
ahh that's why my target+decoys were more than double . I chosed option 1